activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -498,7 +498,41 @@ def test_cache_miss_instrumentation
     ActiveSupport::Notifications.unsubscribe "cache_read.active_support"
   end
 
+  def test_nil_is_significant
+    skip "does not support blank key" if does_not_support_blank_keys?
+
+    @cache.write(nil, "baz")
+    assert_equal "baz", @cache.read(nil)
+
+    @cache.write([nil], "bizz")
+    assert_equal "bizz", @cache.read(nil)
+  end
```

Comment:
I don't think we actually want to support singular blank keys. It's nearly always a mistake/error, unless it's a single-purpose cache that's namespaced—quite uncommon.

So, this is the current behavior, but it's an unaddressed edge case that we probably don't want to lock in with test coverage. We'd be better off raising an exception (with migration path) or documenting it as explicitly undefined.

Reply:
Seems good to add an error. I did that here. Even in the namespaced case, I don't think people want an "empty" key.